profiles/audit_log.py

```diff
@@ -0,0 +1,100 @@
+import json
+import logging
+import threading
+from datetime import datetime, timezone
+
+from django.conf import settings
+
+from .utils import get_current_service, get_current_user, get_original_client_ip
+
+_thread_locals = threading.local()
```

Comment:
You probably have thought this through, but using thread locals seems to be at least rare in a Django context. @mikkokeskinen can also comment, maybe this still is the best option here. Only alternative I can think of is to store the data in the request or response which also seems ugly.

Reply:
Yes, it is rare. But when we do logging this way, we don't have any variables to rely on. As the logging is done in the \_\_init\_\_ of the model.

I'd say this is just a temporary solution and we should think the whole audit logging implementation again.

Reply:
Thread locals were in use in the audit logging system before too, so I followed the convention. Other solutions can of course be considered, but I wouldn't include them in this PR.